packages/@expo/cli/src/start/server/metro/withMetroMultiPlatform.ts

```diff
@@ -227,16 +229,7 @@ export function withExtendedResolver(
             preferNativePlatform: platform !== 'web',
             resolveRequest: undefined,
 
-            // @ts-expect-error
             mainFields,
-            // Passing `mainFields` directly won't be considered (in certain version of Metro)
-            // we need to extend the `getPackageMainPath` directly to
-            // use platform specific `mainFields`.
-            getPackageMainPath(packageJsonPath) {
-              // @ts-expect-error: mainFields is not on type
-              const package_ = context.moduleCache.getPackage(packageJsonPath);
-              return package_.getMain(mainFields);
-            },
```

Comment:
removing the `getPackageMainPath` because it's deprecated from metro.

Reply:
For the sake of supporting older versions (for me testing things), it would be nice to keep this around and just ts-ignore it.